src/parser.c

```diff
@@ -348,12 +426,15 @@ json_object *parser_data_create_object(const char *device_id, uint8_t sensor_id,
 			json_object_new_uint64(knot_value_as_uint64(value)));
 		break;
 	default:
-		goto fail;
+		has_err = true;
+		break;
 	}
 
 	json_object_array_add(json_array, data);
 	json_object_object_add(json_msg, KNOT_JSON_FIELD_DATA, json_array);
 
+	json_str = l_strdup(json_object_to_json_string(json_msg));
+	json_object_put(json_msg);
```

Comment:
I do not think this statement should be executed if the `has_err` flag is set as true.

Reply:
done